obspy/core/trace.py

```diff
@@ -194,6 +207,9 @@ def __setitem__(self, key, value):
                 timediff = float(self.npts - 1) * delta
             self.__dict__['endtime'] = self.starttime + timediff
             return
+        if key == 'component':
+            key = 'channel'
+            value = self.channel[:-1] + str(value)
```

Comment:
I have doubts about that substitution. In the end, the component parameter is quite artificial and this kind magic even if totally reasonable might lead to misunderstandings.

Reply:
I agree with @heavelock, probably best to make the `component` attribute read-only and force the user reset the whole channel code if he/she really desires to do so.

Reply:
I am undecided. Channel code and component are well defined by SEED convention.
I know it is special, because setting the component actually changes the channel. Still, there are other attributes behaving magically.

In the end, making it just read-only is OK with me.

Reply:
If we have a getter, I think having a setter might be fine as well. However, I'm only feeling uneasy about setting with empty string or multiple chars.. that's changing the length of the channel code and those examples in the tests are pretty weird to me, going from `HHZ` to `H` with two times setting an empty channel code.

I would agree to the setter if we add a `len(..) == 1` check in it.

Reply:
OK, I added this check. Do you think it is now reasonable, @heavelock, @d-chambers?

Reply:
@trichter, yes I suppose that is a decent compromise. I would still rather be more explicit in setting the whole channel but I can see why this could be more convenient.

Reply:
Yup, I think its a reasonable solution.

Reply:
OK, seems we can all agree that this is a viable solution/compromise. :+1: